tests/Fixer/ControlStructure/SwitchContinueToBreakFixerTest.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\ControlStructure;
+
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\ControlStructure\SwitchContinueToBreakFixer
+ */
+final class SwitchContinueToBreakFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @param string      $expected
+     * @param null|string $input
+     *
+     * @dataProvider provideTestFixCases
+     */
+    public function testFix($expected, $input = null)
+    {
+        $this->doTest($expected, $input);
+    }
+
+    public function provideTestFixCases()
+    {
+        return [
+            'simple case' => [
+                '<?php
+switch($a) {
+    case 1:
+        echo 1;
+
+        break;
+    case 2:
+        echo 2;
+
+        break;
+    case 3:
+        echo 2;
+
+        continue 2;
+}
+',
+                '<?php
+switch($a) {
+    case 1:
+        echo 1;
+
+        continue;
+    case 2:
+        echo 2;
+
+        continue;
+    case 3:
+        echo 2;
+
+        continue 2;
+}
+',
+            ],
+            'nested do while' => [
+                '<?php
+switch ($a) {
+    case 1:
+        do {
+            switch ($a) {
+                case 1:
+                    do {
+                        switch ($a) {
+                            case 1:
+                                do {
+                                    continue;
+                                } while (false);
+
+                            break;
+                        }
+
+                        continue;
+                    } while (false);
+
+                break;
+            }
+            continue;
+        } while (false);
+
+    break;
+}
+',
+                '<?php
+switch ($a) {
+    case 1:
+        do {
+            switch ($a) {
+                case 1:
+                    do {
+                        switch ($a) {
+                            case 1:
+                                do {
+                                    continue;
+                                } while (false);
+
+                            continue;
+                        }
+
+                        continue;
+                    } while (false);
+
+                continue;
+            }
+            continue;
+        } while (false);
+
+    continue;
+}
+',
+            ],
+            'do not fix cases' => [
+                '<?php
+switch($a) {
+    case 1:
+        while (false) {
+            continue;
+        }
+
+        while (false) continue 2;
```

Comment:
Shouldn't this case be fixed as well? `continue 2` is targeting the parent `switch`. I would even make it a dedicated test case.

Reply:
thanks, fixed!